### PR TITLE
Provide a default implementation of common FuDevice actions

### PR DESCRIPTION
**Pull request by hughsie**
This reduces the amount of boilerplate in each plugin.

    32 files changed, 156 insertions(+), 584 deletions(-)

### CODE REVIEW

File: plugins/dell-dock/fu-dell-dock-hub.c [hunk 1]
```diff
@@ -111,6 +111,7 @@ fu_dell_dock_hub_write_fw (FuDevice *device,
 	}
 
 	/* dock will reboot to re-read; this is to appease the daemon */
+	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
```
Comment (superm1) [inline plugins/dell-dock/fu-dell-dock-hub.c]: Actually the device doesn't restart at the end, it restarts when you unplug it from the system.

File: plugins/dell-dock/fu-dell-dock-i2c-ec.c [hunk 2]
```diff
@@ -839,6 +839,7 @@ fu_dell_dock_ec_write_fw (FuDevice *device,
 		}
 	}
 
+	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
```
Comment (superm1) [inline plugins/dell-dock/fu-dell-dock-i2c-ec.c]: likewise Actually the device doesn't restart at the end, it restarts when you unplug it.

File: plugins/dell-dock/fu-dell-dock-i2c-mst.c [hunk 3]
```diff
@@ -844,6 +844,7 @@ fu_dell_dock_mst_write_fw (FuDevice *device,
 	}
 
 	/* dock will reboot to re-read; this is to appease the daemon */
+	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
```
Comment (superm1) [inline plugins/dell-dock/fu-dell-dock-i2c-mst.c]: Actually the device doesn't restart at the end, it restarts when you unplug it.

File: plugins/dell-dock/fu-dell-dock-i2c-tbt.c [hunk 4]
```diff
@@ -136,6 +136,7 @@ fu_dell_dock_tbt_write_fw (FuDevice *device,
 	}
 
 	/* dock will reboot to re-read; this is to appease the daemon */
+	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
```
Comment (superm1) [inline plugins/dell-dock/fu-dell-dock-i2c-tbt.c]: Actually the device doesn't restart at the end, it restarts when you unplug it.

File: plugins/dell-dock/fu-dell-dock-status.c [hunk 5]
```diff
@@ -90,6 +90,7 @@ fu_dell_dock_status_write (FuDevice *device,
 		return FALSE;
 
 	/* dock will reboot to re-read; this is to appease the daemon */
+	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
```
Comment (superm1) [inline plugins/dell-dock/fu-dell-dock-status.c]: Actually the device doesn't restart at the end, it restarts when you unplug it.

File: plugins/dell-dock/fu-dell-dock-hub.c (hunk 1 above)
Comment (superm1) [reply plugins/dell-dock/fu-dell-dock-hub.c]: nevermind, I understand now

File: plugins/dell-dock/fu-plugin-dell-dock.c [hunk 6]
```diff
@@ -240,20 +215,3 @@ fu_plugin_composite_cleanup (FuPlugin *plugin,
 
 	return fu_dell_dock_ec_reboot_dock (parent, error);
 }
-
-gboolean
-fu_plugin_activate (FuPlugin *plugin, FuDevice *device, GError **error)
-{
-	g_autoptr(FuDeviceLocker) locker = NULL;
-	if (!FU_IS_DELL_DOCK_EC (device)) {
```
Comment (superm1) [inline plugins/dell-dock/fu-plugin-dell-dock.c]: I guess this is lost, but not super important.

Comment (hughsie) [reply plugins/dell-dock/fu-plugin-dell-dock.c]: I checked, the EC is the only one implementing the activate vfunc.

**Pull request merged**